matchms/filtering/require_correct_ionmode.py

```diff
@@ -0,0 +1,27 @@
+import logging
+from matchms import Spectrum
+
+
+logger = logging.getLogger("matchms")
+
+
+def require_correct_ionmode(spectrum_in: Spectrum,
+                            ion_mode_to_keep):
+    """Removes spectra that are not in the correct ionmode"""
+    if spectrum_in is None:
+        return None
+    spectrum = spectrum_in.clone()
+
+    assert ion_mode_to_keep in {"positive", "negative", "both"}, \
+        "ion_mode should be set to 'positive', 'negative' or 'both'"
```

Comment:
assert is generally not advised to use in normal code I think - the recommended thing is to throw an exception. (https://stackoverflow.com/questions/944592/best-practice-for-using-assert) asserts might also get removed for optimization, so they are not a very stable way of handling errors

Reply:
We could also start having a collection of constants as enums or so for these categorical variables.

Reply:
The collection of constants might be good to add at some point indeed. I will leave it like this for now.